plutus-core/plutus-core/src/PlutusCore/Parser/Builtin.hs

```diff
@@ -84,11 +83,15 @@ conPair uniA uniB = trailingWhitespace . inParens $ do
   pure (a, b)
 
 conData :: Parser Data
-conData = do
-  b <- conBS
-  case deserialiseOrFail (Lazy.fromStrict b) of
-    Right d  -> pure d
-    Left err -> getSourcePos >>= customFailure . InvalidData (T.pack (show err))
+conData =
+    choice
+        [ trailingWhitespace $ inParens conData
```

Comment:
Why do we need `trailingWhitespace` here and above?  I thought that whitespace was pretty much dealt with automatically.

Reply:
Oh wait. this allows `(con data (((((I 5))))))`. Why do we want that?

Reply:
> Why do we need `trailingWhitespace` here and above?

It's something related to getting accurate source positions. If we made `inParens` consume trailing whitespace, we'd get inaccurate source position information, so we don't. As far as I understand.

Reply:
> Extra parenthesis never hurt anyone.

I reckon I could find a counter example to that if I tried!  It seems a bit odd if they're allowed inside `Data` but not elsewhere though.  However, it's not a big deal: in this case they do seem harmless.

Reply:
>  However, it's not a big deal: in this case they do seem harmless.

I've already changed the behavior anyway, now we shouldn't accept extra parens, although I didn't write any tests for that as there are no such tests for anything else either (as far as I'm aware).